app/Main.hs

```diff
@@ -22,20 +23,26 @@ data CLI
       (Maybe FilePath) -- file to run
       Bool -- cheat mode
   | Format Input
+  | DocGen GenerateDocs
   | LSP
 
 cliParser :: Parser CLI
 cliParser =
   subparser
     ( command "format" (info (format <**> helper) (progDesc "Format a file"))
         <> command "lsp" (info (pure LSP) (progDesc "Start the LSP"))
+        <> command "generate" (info (DocGen <$> docgen <**> helper) (progDesc "Generate docs"))
```

Comment:
We might want to add the `hidden` mod.

Reply:
Do you mean for the whole subparser? All of the commands show just as COMMAND, so it does not clutter much:
```
Swarm game - pre-alpha version (generate-docs@58cd7c00c3)

Usage: swarm [COMMAND | [-s|--seed INT] [-c|--scenario FILE] [-r|--run FILE] 
               [-x|--cheat]]

  To play the game simply run without any command.

Available options:
  -s,--seed INT            Seed to use for world generation
  -c,--scenario FILE       Name of a scenario to load
  -r,--run FILE            Run the commands in a file at startup
  -x,--cheat               Enable cheat mode
  -h,--help                Show this help text

Available commands:
  format                   Format a file
  lsp                      Start the LSP
  generate                 Generate docs
```

Reply:
But developers will need the `--help` to show something and I do not think that *one line* is worth adding a `--help-all` or conditionally hiding it based on for example Git branch or build flag.

Reply:
I remember having the same discussion for the `lsp` sub command, where I suggested a more minimalist usage where only the relevant option are presented to the user. Though I don't mind the other way, and let's not bother adding an extra `--help-all` :)

Reply:
As long as it fits on a small terminal screen I think it's fine.

Besides, players can now use the menu screen. :smile: